engine/src/main/java/org/terasology/logic/players/MenuControlSystem.java

```diff
@@ -28,19 +28,26 @@
 import org.terasology.input.binds.general.OnlinePlayersButton;
 import org.terasology.input.binds.general.PauseButton;
 import org.terasology.input.events.KeyDownEvent;
+import org.terasology.logic.characters.CharacterComponent;
 import org.terasology.logic.characters.events.DeathEvent;
+import org.terasology.logic.health.DoDestroyEvent;
 import org.terasology.network.ClientComponent;
 import org.terasology.registry.CoreRegistry;
 import org.terasology.registry.In;
 import org.terasology.rendering.nui.NUIManager;
+import org.terasology.rendering.nui.layers.ingame.DeathScreen;
 import org.terasology.rendering.nui.layers.ingame.OnlinePlayersOverlay;
 import org.terasology.rendering.opengl.ScreenGrabber;
+import org.terasology.utilities.Assets;
 
 /**
  */
 @RegisterSystem(RegisterMode.CLIENT)
 public class MenuControlSystem extends BaseComponentSystem {
 
+    private String lastDamage = "Unknown";
+    private String lastInstigator = "Unknown";
```

Comment:
I'm not sure how far i18n is for the game, but "Unknown" should either be a constant reused or a key to a translatable string.